### PR TITLE
Move polymer-build configuration logic to polymer-project-config

### LINKED ISSUE #36: excludeDependencies?
Can we include excludeDependencies option as part of the build? or is there a simpler way to exclude files from the build?

### PR DESCRIPTION
**Pull request by FredKSchott**
~~*Tests are expected to fail until polymer-project-config is published to npm*~~

This PR moves old configuration logic over to the new polymer-project-config library. See https://github.com/Polymer/polymer-project-config/pull/1 for background and bigger discussion on the move.

Fixes #36

/cc @justinfagnani

### CODE REVIEW

File: src/analyzer.ts [hunk 1]
```diff
@@ -165,7 +145,7 @@ export class StreamAnalyzer extends Transform {
   }
 
   isFragment(file: File): boolean {
```
Comment (justinfagnani) [inline src/analyzer.ts]: seems like this should move into `ProjectConfig`

File: src/bundle.ts [hunk 2]
```diff
@@ -101,30 +82,30 @@ export class Bundler extends Transform {
   }
 
   isEntrypoint(file: File): boolean {
```
Comment (justinfagnani) [inline src/bundle.ts]: same here: move into `ProjectConfig`

File: src/bundle.ts [hunk 3]
```diff
@@ -101,30 +82,30 @@ export class Bundler extends Transform {
...
         let newShellContent = this._addSharedImportsToShell(bundles);
         shellFile.contents = new Buffer(newShellContent);
       }
 
-      for (let fragment of this.allFragments) {
-        let fragmentUrl = urlFromPath(this.root, fragment);
-        let addedImports = (fragment === this.shell && this.shell)
+      for (let fragment of this.config.allFragments) {
+        let fragmentUrl = urlFromPath(this.config.root, fragment);
+        let addedImports = (fragment === this.config.shell && this.config.shell)
```
Comment (justinfagnani) [inline src/bundle.ts]: for clarity let's move the `fragment === this.config.shell && this.config.shell` into a variable named `isShell`

File: src/bundle.ts (hunk 2 above)
Comment (FredKSchott) [inline src/bundle.ts]: 👍 great call

File: src/analyzer.ts (hunk 1 above)
Comment (FredKSchott) [inline src/analyzer.ts]: 👍 great call

Comment (FredKSchott) [inline src/analyzer.ts]: this makes the "fragments" vs. "allFragments" collision even more confusing (https://github.com/Polymer/polymer-project-config/pull/1#r79076957)

Comment (justinfagnani) [inline src/analyzer.ts]: Yeah, super not happy with the naming... :(

**Comment by FredKSchott** [comment]
@justinfagnani @rictic @usergenic PTAL

**Comment by rictic** [comment]
Cool! I like the deduplication here. Thanks for ccing me in so I can get some more polymer-build context.

Leaving the review proper to @justinfagnani and @usergenic as they'll have better context right now though.

**Pull request merged**